pkg/proxy/proxier.go

```diff
@@ -548,12 +555,33 @@ func (proxier *Proxier) openPortal(service types.NamespacedName, info *serviceIn
 	if err != nil {
 		return err
 	}
+
+	publicIpCount := 0
 	for _, publicIP := range info.publicIP {
-		err = proxier.openOnePortal(net.ParseIP(publicIP), info.portalPort, info.protocol, proxier.listenIP, info.proxyPort, service)
+		ip := net.ParseIP(publicIP)
+		if ip == nil {
+			glog.V(2).Info("Skipping non-address public endpoint: %s", publicIP)
+			continue
+		}
+		publicIpCount++
+		err = proxier.openOnePortal(ip, info.portalPort, info.protocol, proxier.listenIP, info.proxyPort, service)
 		if err != nil {
 			return err
 		}
 	}
+
+	// We have an external load balancer, but no specific IPs.
```

Comment:
I'm wondering if it is time to fall into calling a CloudProvider API, rather than try to build a very cumbersome set of rules  into the values in PublicIPs.

Just so I understand correctly - as implemented this will have a global limit of one Service per port.  Period.  Once someone uses port 80 for an external service, nobody else can use it?

I am out Thursday and Friday, but I'd like to chew on this a bit more.  There has to be a better model.  Can we maybe do an IRC discussion or a hangout next week to discuss?  It may be that is the best we can do for now, but if so, I want to understand where we can go with it after 1.0.

@satnam6502 since we were discussing this today.